### DISCUSSION TITLE
How to build flow with a block thats only to be executed after the For Each block

### DISCUSSION THREAD

**Discussion by gettosam** [postmanlabs/postman-flows#297] [Q&A]
I have lot of blocks i put inside the for loop. and I want to execute some more block, but ONLY after the for loop ends.

how do i do this in postman flow ? the small checkbox in For => it says trigger when the block finishes, but it does not carry any data, and it also shows in disabled color everytime

**Comment by samay-v** [comment]
You can achieve this by using the signal ports right now. Each block has "Signal ports" that triggers once the block has finished running. If you attach the signal-out of the `For each` to the signal-in of another block, the other block will trigger after the "Loop" ends. Remember though, the signal port does not carry any data 

![image](https://user-images.githubusercontent.com/28357106/163023396-b74e928e-9b10-4a16-8c7e-a63d0dcc8a95.png)

**Reply by gettosam** [reply]
Thanks,  but in this case, i lose all the Durables data, as well as the previous for records data right in "Runs after the loop is completed" block ?

The Off button says, "Triggers the block but carries no data "

**Reply by gettosam** [reply]
I do have some Initial Data configuration in "Create Durable" block, thats where i configure all initial variables, that i want to carry out all the way to the end.

atleast i need the Durables data in this after For each finishes.
